test/src/androidTest/kotlin/io/realm/NullabilityTests.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.runtimeapi.RealmModule
+import test.Nullability
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class NullabilityTests {
+
+    @RealmModule(Nullability::class)
+    class MySchema
+
+    lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        val configuration = RealmConfiguration.Builder(schema = MySchema()).build()
+        realm = Realm.open(configuration)
+        // FIXME Cleaning up realm to overcome lack of support for deleting actual files
+        //  https://github.com/realm/realm-kotlin/issues/95
+        realm.beginTransaction()
+        realm.objects(Nullability::class).delete()
+        realm.commitTransaction()
+        assertEquals(0, realm.objects(Nullability::class).size, "Realm is not empty")
+    }
+
+    @Test
+    fun nullability() {
+        realm.beginTransaction()
+        val nullability = realm.create(Nullability::class)
+        assertNull(nullability.stringNullable)
+        // What is the default semantics for this? Can we fix it before default values
+        assertNull(nullability.stringNullable)
+
+        nullability.stringNullable = "Realm"
+        assertNotNull(nullability.stringNullable)
+        nullability.stringNullable = null
+        assertNull(nullability.stringNullable)
+
+        // Should we try to verify that compiler will break on this
+        // nullability.stringNonNullable = null
```

Comment:
What would be the benefit of this test at this layer (since the user cannot write such code?) we could check at the c-interop level the behaviour of passing null on a non-nullable column

Reply:
Yeah. It was to somehow have a test that ensures that we don't mess up the checks for nullability now that I implemented the setters/getters as generic methods. Maybe it is at the wrong place, but it is just so damn easy to write the test when the schema is generated (which requires the compiler plugin). I will see if I can come up with a good pattern.

Reply:
Come to think of it. I guess `RealmObjectHelper` should in fact be in `library`, which would allow us to do the out-commented test, but I didn't want to do that in this PR as it ties a bit into the origanization of `@RealmObject` vs `RealmModel`-interface and similar.

Reply:
I created an issue for reconsidering the module structure in #134. Let's do the clean up there.